core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
I like the idea of being able reuse validators, but I think we're going to run into the same problem we're seeing in this pull request. How do we get information in the error? In the example above, if one of the RequestReaders fails to validate, how do we know which one it is? We're already having some difficulty with getting the Readers to provide the information we want.